client/src/components/TournamentAdminPanel.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useReducer, Dispatch, SetStateAction } from 'react'
+import DeleteForeverIcon from '@material-ui/icons/DeleteForever'
+import EditIcon from '@material-ui/icons/Edit'
+import {
+  Container,
+  Typography,
+  Button,
+  Divider,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  makeStyles,
+  Theme,
+  createStyles,
+} from '@material-ui/core'
+import { isAdmin } from '../hooks/useUsers'
+import { useCurrentUser } from '../hooks/useCurrentUser'
+import { Tournament } from '../hooks/useTournaments'
+import { MessageSnackBar } from './MessageSnackBar'
+import { useHistory } from 'react-router-dom'
+import { request } from '../utils/request'
+import { EditTournamentModal } from '../modals/EditTournamentModal'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      position: 'relative',
+      minHeight: theme.spacing(10),
+    },
+    button: {
+      margin: theme.spacing(1),
+      bottom: theme.spacing(2),
+      float: 'right',
+    },
+  })
+)
+
+interface State {
+  dialogOpen: boolean
+  snackBarOpen: boolean
+  requestError: boolean
+  snackBarMessage: string
+  editModalOpen: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function reducer(state: State, action: any) {
+  switch (action.type) {
+    case 'OPEN_DIALOG':
+      return { ...state, dialogOpen: true }
+    case 'CLOSE_DIALOG':
+      return { ...state, dialogOpen: false }
+    case 'OPEN_EDIT_MODAL':
+      return { ...state, editModalOpen: true }
+    case 'CLOSE_EDIT_MODAL':
+      return { ...state, editModalOpen: false }
+    case 'UPDATE_SUCCESS':
+      return {
+        ...state,
+        editModalOpen: false,
+        snackBarMessage: action.payload,
+        requestError: false,
+        snackBarOpen: true,
+      }
+    case 'REQUEST_ERROR':
+      return {
+        ...state,
+        snackBarOpen: true,
+        snackBarMessage: action.payload,
+        requestError: true,
+      }
+    case 'CLOSE_SNACKBAR':
+      return {
+        ...state,
+        snackBarOpen: false,
+      }
+    default:
+      throw new Error()
+  }
+}
+
+export function TournamentAdminPanel(props: {
+  tournament: Tournament
+  setTournament: Dispatch<SetStateAction<Tournament | undefined>>
```

Comment:
This is the callback triggered when tournament is updated successfully. It would probably be better names `onTournamentUpdate` or something like that.